### DISCUSSION TITLE
Add a minimalist splash screen to empty.html

### DISCUSSION THREAD

**Discussion by AnthonyMuscio** [TiddlyWiki/TiddlyWiki5#5827] [Ideas]
I have currently got the unhappy situation where my broadband provider has failed to move my broadband to my new address. Broken beaurocracy is on of my biggest frustrations. 

What this means is I am running at less than 1MB internet access, and I thought I would share the fact that many tiddlywikis look like they are broken when no loading splash screen is available.

So I would like to suggest not only Add a splash screen to large wikis, a must, but perhaps even add them to all internet published wikis.

I have raised this in the [google groups](https://groups.google.com/g/tiddlywiki/c/g1fyKM38kG4) but raise it here that perhaps we should **Add a minimalist splash screen to empty.html** because it in TiddlyWikis interest to not appear to be broken.

Perhaps we could say something like "Loading your interactive TiddlyWiki, please wait!" so as to hint at the fact that a fully loaded wiki is a high performance interactive environment.

Regards
Tones

**Comment by pmario** [comment]
How long is your "white screen" with https://tiddlywiki.com/empty.html? 
With shift - page reload and a standard page reload?

**Comment by AnthonyMuscio** [comment]
It would take a while at 500kbps but once made into a wiki with user content it can be of any size and over any speed.

**Reply by pmario** [reply]
In the OP you wrote 1MB which is 1MByte = 10MBit/s for me. That's why I was curious. But 500kbit is slow. ... 

I would support the request. IMO it doesn't hurt for fast connections, because users won't see it. And where we have a slower connection it helps the wiki.

**Comment by Jermolene** [comment]
Hi @AnthonyMuscio it's a reasonable suggestion, but as it turns out rather surprisingly hard to implement:

* There's no way for the core to tell whether a splash screen is installed (splash screens are just rawmarkup tiddlers that include content with the class `tc-remove-when-wiki-loaded`), so we can't have a fallback if the user doesn't specify a splash screen
* If we ship a splash screen in empty.html then it would have to be as an additional system tiddler. The trouble is that users wouldn't have an obvious way to figure out how to remove the built-in splash screen if they wanted to add their own

**Reply by AnthonyMuscio** [reply]
Jeremy, I understand your concern but perhaps it could be an option in the basic settings, a checkbox for default splashscreen (toggle the raw markup tag), with a few links to also support customising it.

I have a shapshot tool that captures the html result of any tiddler as html, or it can be copied from the internals html view and pasted into the splash screen tiddler. Sometimes the inclusion of small additional features have a great impact on users by demonstrating how it can be done, part of the self documenting nature of tiddlywiki.

The point would be the only people that may have a problem is someone adding their own splash screen, and at that time they would be reading the doco to see how to do it, where we can document disabling the default.

Regards
Tones

**Comment by AnthonyMuscio** [comment]
Post script, this is in my skill scope so I could produce it and do a PR?
